src/main/kotlin/nl/info/client/brp/util/BRPClientHeadersFactory.kt

```diff
@@ -4,21 +4,73 @@
  */
 package nl.info.client.brp.util
 
+import jakarta.enterprise.inject.Instance
+import jakarta.enterprise.inject.UnsatisfiedResolutionException
+import jakarta.inject.Inject
 import jakarta.ws.rs.core.MultivaluedMap
-import org.eclipse.microprofile.config.ConfigProvider
+import nl.info.zac.authentication.LoggedInUser
+import org.eclipse.microprofile.config.inject.ConfigProperty
 import org.eclipse.microprofile.rest.client.ext.ClientHeadersFactory
+import java.util.Optional
+import java.util.logging.Level
+import java.util.logging.Logger
+
+class BRPClientHeadersFactory @Inject constructor(
+    @ConfigProperty(name = "brp.api.key")
+    private val apiKey: Optional<String> = Optional.empty(),
+
+    @ConfigProperty(name = "brp.origin.oin")
```

Comment:
We should define these properties or not? In our env.tpl, env.example, Docker Compose, itests, Helm Chart?

Reply:
Yes, for sure. That's why this is still a draft. Also looking for a nice way to test these.

My idea was to have them optional, but if these are the default values you're right that it doesn't make sense. Mandatory defaults would be more useful.

Reply:
Property/variable is mandatory now